dbms/src/Storages/MergeTree/registerStorageMergeTree.cpp

```diff
@@ -637,6 +637,9 @@ static StoragePtr create(const StorageFactory::Arguments & args)
         throw Exception("You must set the setting `allow_experimental_data_skipping_indices` to 1 " \
                         "before using data skipping indices.", ErrorCodes::BAD_ARGUMENTS);
 
+    /// Finalize settings and disable updates
+    storage_settings->finishSettingsInitialization();
```

Comment:
I think it would be clearer to directly set the fields to be immutable here:

```
storage_settings->index_granularity_bytes.setReadOnly(true);
storage_settings->index_granularity.setReadOnly(true);
```

Reply:
How will we know which settings are immutable? Iterate through all of them?